frontend/src/components/pages/volunteer/posting/VolunteerPostingsPage.tsx

```diff
@@ -1,11 +1,240 @@
+//     autoClosingDate: new Date("Monday, November 30"),
+//     isSignedUp: false,
+//     description:
+//       "Volunteers will be responsible for updating the inventory monthly, updating the manuals and guidelines on an ongoing basis, uploading the COVID-19 screenings on a weekly basis.",
+//   },
+//   {
+//     id: "2",
+//     skills: [
+//       { id: "1", name: "CPR" },
+//       { id: "2", name: "WHMIS" },
+//       { id: "3", name: "Photocopying" },
+//     ],
+//     branch: {
+//       id: "1",
+//       name: "sample text",
+//     },
+//     title: "Medical Reception Volunteer",
+//     startDate: new Date("February 14, 2022"),
+//     endDate: new Date("March 20, 2022"),
+//     autoClosingDate: new Date("Monday, November 30"),
+//     isSignedUp: false,
+//     description:
+//       "Volunteers will be responsible for updating the inventory monthly, updating the manuals and guidelines on an ongoing basis, uploading the COVID-19 screenings on a weekly basis.",
+//   },
+//   {
+//     id: "3",
+//     skills: [
+//       { id: "1", name: "CPR" },
+//       { id: "2", name: "WHMIS" },
+//       { id: "3", name: "Photocopying" },
+//     ],
+//     branch: {
+//       id: "1",
+//       name: "sample text",
+//     },
+//     title: "Medical Reception Volunteer",
+//     startDate: new Date("March 18, 2022"),
+//     endDate: new Date("March 20, 2022"),
+//     autoClosingDate: new Date("Monday, November 30"),
+//     isSignedUp: false,
+//     description:
+//       "Volunteers will be responsible for updating the inventory monthly, updating the manuals and guidelines on an ongoing basis, uploading the COVID-19 screenings on a weekly basis.",
+//   },
+// ];
+
+// can refactor this query in the future to be more specific (depending on the data this page needs)
+const POSTINGS = gql`
+  query VolunteerPostingsPage_postings {
+    postings {
+      id
+      branch {
+        id
+        name
+      }
+      shifts {
+        id
+        postingId
+        startTime
+        endTime
+      }
+      skills {
+        id
+        name
+      }
+      employees {
+        id
+        branchId
+      }
+      title
+      type
+      status
+      description
+      startDate
+      endDate
+      autoClosingDate
+      numVolunteers
+    }
+  }
+`;
+
+const MS_PER_WEEK = 604800000;
 
 const VolunteerPostingsPage = (): React.ReactElement => {
+  const [postings, setPostings] = useState<PostingResponseDTO[] | null>(null);
+  const [unfilteredPostings, setUnfilteredPostings] = useState<
+    PostingResponseDTO[] | null
+  >(null);
+  const [filter, setFilter] = useState<string>("week");
+
+  useQuery(POSTINGS, {
+    fetchPolicy: "cache-and-network",
+    onCompleted: (data) => {
+      setPostings(data.postings);
+      setUnfilteredPostings(data.postings);
+    },
+  });
+
+  // should posting.startDate type be declared as string instead on frontend?
+  const dateCompareUtil = (start: Date, filterType: string): boolean => {
```

Comment:
```suggestion
  const dateInRange = (start: Date, filterType: string): boolean => {
```

Also, since this function will likely be reused for other filter components, it'd be a good idea to add this to `DateUtils.ts` 🙂